packages/widgetbook/lib/src/fields/iterable_segmented_field.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/material.dart';
+
+import 'fields.dart';
+
+/// A [Field] that builds [SegmentedButton]<[Iterable<T>]> for [Iterable] values.
+class IterableSegmentedField<T> extends Field<Iterable<T>> {
+  /// Creates a new instance of [IterableSegmentedField].
+  IterableSegmentedField({
+    required super.name,
+    required this.values,
+    required super.initialValue,
+    this.labelBuilder = defaultLabelBuilder,
+    this.emptySelectionAllowed = true,
+    @Deprecated('Fields should not be aware of their context') super.onChanged,
+  }) : super(
+         defaultValue: values,
+         type: FieldType.iterableSegmented,
+         codec: FieldCodec(
+           toParam: (value) => "[${value.map(labelBuilder).join(',')}]",
+           toValue: (param) {
+             if (param == null) return null;
+             if (!param.startsWith('[') || !param.endsWith(']')) return [];
```

Comment:
This case should return `null` (i.e. unparsable), since it's invalid format, not an empty iterable.

When `null` is returned, we fall back to things like default values.
When `[]` this means it will return empty Iterable, even if the initial value is different.